airbyte-cdk/python/airbyte_cdk/sources/declarative/auth/oauth.py

```diff
@@ -0,0 +1,101 @@
+#
+# Copyright (c) 2022 Airbyte, Inc., all rights reserved.
+#
+
+from typing import Any, List, Mapping, MutableMapping, Tuple
+
+import pendulum
+import requests
+from airbyte_cdk.sources.declarative.interpolation.interpolated_mapping import InterpolatedMapping
+from airbyte_cdk.sources.declarative.interpolation.interpolated_string import InterpolatedString
+from requests.auth import AuthBase
+
+
+class Oauth2Authenticator(AuthBase):
```

Comment:
as per some offline discussion, we're now going to have a separate declarative oauth authenticator. Its annoying that we have a declarative oauth authenticator that looks roughly similar to the existing CDK one. But at least this can support runtime interpolation w/o shoe horning it into the existing implementation.